__tests__/integration.js

```diff
@@ -0,0 +1,391 @@
+      expect(response).toEqual(makeResponse({
+        "body": `[{"id":2,"name":"Jane"},{"id":3,"name":"${newName}"}]`,
+        "headers": {
+          "content-length": "68",
+          "etag": "W/\"44-AtuxlvrIBL8NXP4gvEQTI77suNg\"",
+        },
+        statusCode: 200
+      }))
+      done()
+    }
+    lambdaFunction.handler(makeEvent({
+      path: '/users/1',
+      httpMethod: 'DELETE'
+    }), {
+      succeed
+    })
+  })
+
+  test('PUT JSON', (done) => {
+    const succeed = response => {
+      delete response.headers.date
+      expect(response).toEqual(makeResponse({
+        "body": '{"id":2,"name":"Samuel"}',
+        "headers": {
+          "content-length": "24",
+          "etag": "W/\"18-uGyzhJdtXqacOe9WRxtXSNjIk5Q\"",
+        },
+        statusCode: 200
+      }))
+      done()
+    }
+    lambdaFunction.handler(makeEvent({
+      path: '/users/2',
+      httpMethod: 'PUT',
+      body: '{"name": "Samuel"}'
+    }), {
+      succeed
+    })
+  })
+
+  test('base64 encoded request', (done) => {
+    const succeed = response => {
+      delete response.headers.date
+      expect(response).toEqual(makeResponse({
+        "body": '{"id":2,"name":"Samuel"}',
+        "headers": {
+          "content-length": "24",
+          "etag": "W/\"18-uGyzhJdtXqacOe9WRxtXSNjIk5Q\"",
+        },
+        statusCode: 200
+      }))
+      done()
+    }
+    lambdaFunction.handler(makeEvent({
+      path: '/users/2',
+      httpMethod: 'PUT',
+      body: btoa('{"name": "Samuel"}'),
+      isBase64Encoded: true
+    }), {
+      succeed
+    })
+  })
+
+  test('forwardConnectionErrorResponseToApiGateway', (done) => {
+    const succeed = response => {
+      delete response.headers.date
+      expect(response).toEqual({
+        "body": "",
+        "headers": {},
+        statusCode: 502
+      })
+      done()
+    }
+    lambdaFunction.handler(makeEvent({
+      path: '/',
+      httpMethod: 'GET',
+      body: "{\"name\": \"Sam502\"}",
+      headers: {
+        'Content-Length': '-1'
+      }
+    }), {
+      succeed
+    })
+  })
+  
+  const mockApp = function (req, res) {
+    res.end('')
+  }
+
+  test('forwardLibraryErrorResponseToApiGateway', (done) => {
+    const succeed = response => {
+      expect(response).toEqual({
+        statusCode: 500,
+        body: '',
+        headers: {}
+      })
+      done()
+    }
+    awsServerlessExpress.proxy(server, null, {
+      succeed
+    })
+  })
+
+  test('serverListenCallback', (done) => {
+    const serverListenCallback = jest.fn()
+    const serverWithCallback = awsServerlessExpress.createServer(mockApp, serverListenCallback)
+    const succeed = response => {
+      expect(response.statusCode).toBe(200)
+      expect(serverListenCallback).toHaveBeenCalled()
+      serverWithCallback.close()
+      done()
+    }
+    awsServerlessExpress.proxy(serverWithCallback, makeEvent({}), {
+      succeed
+    })
+  })
+
+  test('server.onError EADDRINUSE', (done) => {
```

Comment:
This error/edge case is now *incredibly* unlikely since we use a random string for the sock path.